Midichlorians/Dev/Beta2/iOS/MidichloriansImpl.mm

```diff
@@ -0,0 +1,357 @@
+        preferredLocalizations += [loc UTF8String] + std::string(" ");
+    }
+    
+    NSLocale *locale = [NSLocale currentLocale];
+    std::string userInterfaceIdiom = "phone";
+    if (device.userInterfaceIdiom == UIUserInterfaceIdiomPad) {
+        userInterfaceIdiom = "pad";
+    } else if (device.userInterfaceIdiom == UIUserInterfaceIdiomUnspecified) {
+        userInterfaceIdiom = "unspecified";
+    }
+    
+    std::map<std::string, std::string> info = {
+        {"deviceName", UnsafeToStdString(device.name)},
+        {"deviceSystemName", UnsafeToStdString(device.systemName)},
+        {"deviceSystemVersion", UnsafeToStdString(device.systemVersion)},
+        {"deviceModel", UnsafeToStdString(device.model)},
+        {"deviceUserInterfaceIdiom", userInterfaceIdiom},
+        {"screens", std::to_string([UIScreen screens].count)},
+        {"screenBounds", RectToString(screen.bounds)},
+        {"screenScale", std::to_string(screen.scale)},
+        {"preferredLanguages", preferredLanguages},
+        {"preferredLocalizations", preferredLocalizations},
+        {"localeIdentifier", UnsafeToStdString([locale objectForKey:NSLocaleIdentifier])},
+        {"calendarIdentifier", UnsafeToStdString([[locale objectForKey:NSLocaleCalendar] calendarIdentifier])},
+        {"localeMeasurementSystem", UnsafeToStdString([locale objectForKey:NSLocaleMeasurementSystem])},
+        {"localeDecimalSeparator", UnsafeToStdString([locale objectForKey:NSLocaleDecimalSeparator])},
+    };
+    if (device.systemVersion.floatValue >= 8.0) {
+        info.emplace("screenNativeBounds", RectToString(screen.nativeBounds));
+        info.emplace("screenNativeScale", std::to_string(screen.nativeScale));
+    }
+    
+    if (NSClassFromString(@"ASIdentifierManager")) {
+        ASIdentifierManager *manager = [ASIdentifierManager sharedManager];
+        info.emplace("isAdvertisingTrackingEnabled", manager.isAdvertisingTrackingEnabled ? "YES" : "NO");
+        if (manager.advertisingIdentifier) {
+            info.emplace("advertisingIdentifier", UnsafeToStdString(manager.advertisingIdentifier.UUIDString));
+        }
+    }
+    
+    NSURL *url = [options objectForKey:UIApplicationLaunchOptionsURLKey];
+    if (url) {
+        info.emplace("UIApplicationLaunchOptionsURLKey", UnsafeToStdString([url absoluteString]));
+    }
+    NSString *source = [options objectForKey:UIApplicationLaunchOptionsSourceApplicationKey];
+    if (source) {
+        info.emplace("UIApplicationLaunchOptionsSourceApplicationKey", UnsafeToStdString(source));
+    }
+    
+    if (!info.empty()) {
+        [MidichloriansImpl emit:iOSDeviceInfo(info)];
+    }
+#else
+    static_cast<void>(options);
+#endif  // (TARGET_OS_IPHONE > 0)
+}
+
++ (void)emit:(const iOSBaseEvent &)event {
+    Stats &instance = Singleton<Stats>::Instance();
+    Variant<T_IOS_EVENTS> v;
+    EventsVariantDispatchedAssigner assigner(v);
+    current::metaprogramming::RTTIDynamicCall<T_IOS_EVENTS>(event, assigner);
```

Comment:
This syntax doesn't work - it tries to put an object of `iOSBaseEvent` type into Variant and fails to compile due to `iOSBaseEvent` missing in the type list.